### PR TITLE
Fix incorrect number formatting in prettifyNumberOfDownloads function

### PR DESCRIPTION
**Pull request by adamreng**
### Description:

Fixed a bug in the "prettifyNumberOfDownloads" function where numbers between 100,000 and 999,999 were incorrectly formatted. For example, 100,000 was formatted as "1k" instead of "100k". Also added a new unit test class to cover test cases for that function.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Marketplace/Plugins.php [hunk 1]
```diff
@@ -430,14 +430,14 @@ private function addPluginCoverImage(&$plugin): void
      * @param $plugin
      * @return void
      */
-    private function prettifyNumberOfDownloads(&$plugin): void
+    protected function prettifyNumberOfDownloads(&$plugin): void
```
Comment (michalkleiner) [inline plugins/Marketplace/Plugins.php]: Is this visibility change needed? I see in the test you set the accessibility to true anyway.

**Comment by michalkleiner** [comment]
Hey @araichyk, thanks for the PR! Good find re. the issue with the number prettification! Just one minor question on the method visibility.

**Comment by araichyk** [comment]
> Hey @araichyk, thanks for the PR! Good find re. the issue with the number prettification! Just one minor question on the method visibility.

Oh yeah, good one @michalkleiner. Reverted it back to private, thanks!

File: plugins/Marketplace/Plugins.php (hunk 1 above)
Comment (adamreng) [reply plugins/Marketplace/Plugins.php]: Oh yeah, good one @michalkleiner. Reverted it back to private, thanks!

**Review by michalkleiner — APPROVED** [review]

**Comment by michalkleiner** [comment]
Thanks @araichyk, good work!

**Pull request merged**